modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/date/DateDDMFormFieldValueRenderer.java

```diff
@@ -56,50 +55,45 @@
 	public String render(DDMFormFieldValue ddmFormFieldValue, Locale locale) {
 		Value value = ddmFormFieldValue.getValue();
 
-		return render(value.getString(locale), locale);
+		return _render(locale, value.getString(locale));
 	}
 
-	protected String render(String valueString, Locale locale) {
-		if (Validator.isNotNull(valueString)) {
-			try {
-				SimpleDateFormat simpleDateFormat =
-					(SimpleDateFormat)DateFormat.getDateInstance(
-						DateFormat.SHORT,
-						Optional.ofNullable(
-							LocaleThreadLocal.getThemeDisplayLocale()
-						).orElse(
-							locale
-						));
-
-				String pattern = simpleDateFormat.toPattern();
-
-				if (StringUtils.countMatches(pattern, "d") == 1) {
-					pattern = StringUtil.replace(pattern, 'd', "dd");
-				}
-
-				if (StringUtils.countMatches(pattern, "M") == 1) {
-					pattern = StringUtil.replace(pattern, 'M', "MM");
-				}
-
-				if (StringUtils.countMatches(pattern, "y") == 2) {
-					pattern = StringUtil.replace(pattern, 'y', "yy");
-				}
-
-				Format format = FastDateFormatFactoryUtil.getSimpleDateFormat(
-					pattern);
-
-				return format.format(
-					DateUtil.parseDate("yyyy-MM-dd", valueString, locale));
-			}
-			catch (ParseException parseException) {
-				_log.error("Unable to parse date", parseException);
-			}
+	private String _render(Locale defaultLocale, String valueString) {
```

Comment:
@carolmariaabb, I believe you approved the pull before this change. Could you please give your notes on this? Thanks!